src/main/java/org/apache/commons/lang3/RegExUtils.java

```diff
@@ -751,4 +755,67 @@ private static String toStringOrNull(final CharSequence text) {
     public RegExUtils() {
         // empty
     }
+
+    /**
+     * Finds all matches in the given text according to the specified pattern.
+     *
+     * @param text    the text to search for matches
+     * @param pattern the pattern to search for
+     * @return an unmodifiable list of found matches; returns an empty List if text or pattern is null
+     */
+    public List<String> findMatches(CharSequence text, Pattern pattern) {
```

Comment:
```suggestion
    public static List<String> findAll(CharSequence text, Pattern pattern) {
```

This is an utility class, so all the methods must be `static`.

_Nitpick_: Naming is certainly on of the hardest tasks. Since all the methods in this class deal with matching and have names like `replaceAll`, `removeAll`, I would call this method `findAll`.